### DISCUSSION TITLE
Fetching artifacts from jobs that upload a non-archive file type

### DISCUSSION THREAD

**Discussion by 1riggs** [python-gitlab/python-gitlab#2437] [Q&A]
I have a script to download all the latest artifacts for some project set. I can retrieve any job artifact with an archive artifact, but most of the GitLab-provided jobs (SAST, Code Quality, etc.) do not upload an archive, just the plain file. I noticed that when downloading these artifacts for the GUI the URL includes the query parameter `?file_type=sast`. Is there a way to include the `file_type` query parameter to `project.artifacts.download` that I am missing? Or is there some other way to fetch such an artifact?

**Comment by nejch** [comment]
@1riggs I'm not sure the GitLab API actually provides an option to download these artifacts via the `/api/v4/` endpoints, without explicitly providing them in `paths:`. What is the exact URL you use in the GUI?

**Reply by 1riggs** [reply]
After some conversations in the Gitter chat, I arrived at the same conclusion, that the GitLAB API does not expose those artifacts via the `/api/v4` endpoints. I ended up just modifying the artifacts section of the pipeline definition to ensure an archive file is created and uploaded, which allowed me to retrieve them from the API.

**Reply by nejch** [reply]
Alright, yeah I think that's a good workaround until GitLab implements it. :)
